common/predictive-text/worker/model-compositor.ts

```diff
@@ -217,8 +216,11 @@ class ModelCompositor {
       // Combine duplicate samples.
       let displayText = prediction.sample.displayAs;
 
-      if(displayText == keepOptionText) {
+      if(displayText == keepOptionText || lexicalModel.toKey && displayText == lexicalModel.toKey(keepOptionText)) {
```

Comment:
I feel like this knows a little _too_ much about how a model is supposed to specify `displayAs` and how  that's supposed to behave. Just a note, but I feel like this may lead to bugs in the future.

Reply:
So... should we add a `key` field internally, in case we later add extra 'decorations' to `displayAs`?  The, uh, 'key' is that we need a way to check if a suggestion matches the 'keyed' version of what the user actually typed.

Reply:
Too many keys! 😂 

Maybe it's best to leave this improvement for a future PR 😅

Reply:
I'd like to see extra parentheses on this test as we are mixing `||` and `&&` in the same test, which is easy to get confused. 😉